packages/waku/src/lib/middleware/types.ts

```diff
@@ -16,6 +16,7 @@ export type HandlerContext = {
   readonly context: Record<string, unknown>;
   readonly data: Record<string, unknown>;
   unstable_errs?: unknown[];
+  unstable_onError?: (err: unknown) => void;
```

Comment:
Hmmm, I don't prefer this. At least, we don't need both. But still, `ctx` doesn't feel the right place for "callbacks".

Reply:
Yes. I wasn't really sure where to put it. I mainly just wanted to provide an example for you. Feel free to close and take it from here. Thank you!

Reply:
I'll have a closer look.

Reply:
I just updated #1280. Feel free to send another draft PR if you like.